storage/replica_test.go

```diff
@@ -2618,6 +2626,118 @@ func TestRangeDanglingMetaIntent(t *testing.T) {
 	}
 }
 
+// TestRangeLookupUseReverseScan verifies the correctness of the results which are retrived
+// from RangeLookup by using ReverseScan.
+func TestRangeLookupUseReverseScan(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	// Test ranges: ["a","c"), ["c","f"), ["f","h") and ["h","y").
+	testRanges := []proto.RangeDescriptor{
+		{RangeID: 2, StartKey: proto.Key("a"), EndKey: proto.Key("c")},
+		{RangeID: 3, StartKey: proto.Key("c"), EndKey: proto.Key("f")},
+		{RangeID: 4, StartKey: proto.Key("f"), EndKey: proto.Key("h")},
+		{RangeID: 5, StartKey: proto.Key("h"), EndKey: proto.Key("y")},
+	}
+	// The range ["f","h") has dangling intent in meta2.
+	withIntentRangeIndex := 2
+
+	txn := &proto.Transaction{ID: uuid.NewUUID4(), Timestamp: tc.clock.Now()}
+	for i, r := range testRanges {
+		if i != withIntentRangeIndex {
+			// Write the new descriptor as an intent.
+			data, err := gogoproto.Marshal(&r)
+			if err != nil {
+				t.Fatal(err)
+			}
+			pArgs := putArgs(keys.RangeMetaKey(r.EndKey), data,
+				tc.rng.Desc().RangeID, tc.store.StoreID())
+			pArgs.Txn = txn
+
+			if _, err := tc.rng.AddCmd(tc.rng.context(), &pArgs); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
+	// Resolve the intents.
+	rArgs := &proto.ResolveIntentRangeRequest{
+		RequestHeader: proto.RequestHeader{
+			Timestamp: txn.Timestamp,
+			Key:       keys.RangeMetaKey(proto.Key("a")),
+			EndKey:    keys.RangeMetaKey(proto.Key("z")),
+			RangeID:   tc.rng.Desc().RangeID,
+			Replica:   proto.Replica{StoreID: tc.store.StoreID()},
+			Txn:       txn,
+		},
+	}
+	rArgs.Txn.Status = proto.COMMITTED
+	if _, err := tc.rng.AddCmd(tc.rng.context(), rArgs); err != nil {
+		t.Fatal(err)
+	}
+
+	// Test ReverseScan without intents.
+	verifyResultsReturnedFromReverseScan(t, tc, testRanges)
+
+	// Write the new descriptor as an intent.
+	intentRange := testRanges[withIntentRangeIndex]
+	data, err := gogoproto.Marshal(&intentRange)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pArgs := putArgs(keys.RangeMetaKey(intentRange.EndKey), data, 1, tc.store.StoreID())
+	pArgs.Txn = &proto.Transaction{ID: uuid.NewUUID4(), Timestamp: tc.clock.Now()}
+	pArgs.Timestamp = pArgs.Txn.Timestamp
+	if _, err := tc.rng.AddCmd(tc.rng.context(), &pArgs); err != nil {
+		t.Fatal(err)
+	}
+
+	// Test ReverseScan with intents.
+	verifyResultsReturnedFromReverseScan(t, tc, testRanges)
+}
+
+func verifyResultsReturnedFromReverseScan(t *testing.T, tc testContext, testRanges []proto.RangeDescriptor) {
```

Comment:
it's better if you merge this method into the actual test since it contains test data. That way the test itself will have more code, but you can group the expected data at the top and explain why they are expected.